src/stores/utils.js

```diff
@@ -0,0 +1,37 @@
+import EventEmitter from "events";
+import { writable } from "svelte/store";
+
+const object_equal = (a, b) => {
+    if (!a || !b) return false;
+    for (const key in a) {
+        if (a[key] !== b[key]) {
+            return false;
+        }
+    }
+    return true;
+};
+
+export const makeStore = (t, idField = "UAR") => {
+    const _e = new EventEmitter();
+    const _s = writable(t);
+    const { subscribe, set, update } = _s;
+
+    const push = (media, emit = true) => _s.update(prev => {
+        if (!prev) return { [media[idField]]: media };
+
+        if (object_equal(prev[media[idField]], media)) {
+            return;
```

Comment:
Will this cause an error on the push event listener as we just return (we are using `media.rows` and `media.content_analysis` in the push event listener)? Could we also have an error in the svelte update process, as docs say it should _return the new value to be set to the store_?

Reply:
it wont, it will only not update the store. see svelte store contract for more information.

Reply:
the push event listener is not called if we exit here.

Reply:
true story